### PR TITLE
[SPARK-11260][SPARKR] with() function support

### PR DESCRIPTION
**Pull request by adrian555**

### CODE REVIEW

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -2045,3 +2045,34 @@ setMethod("attach",
...
+#' @examples
+#' \dontrun{
+#' with(irisDf, nrow(Sepal_Width))
+#' }
+#' @seealso \link{attach}
+setMethod("with",
+          signature(data = "DataFrame"),
+          function(data, expr, ...) {
+            stopifnot(!missing(data))
+            stopifnot(inherits(data, "DataFrame"))
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: no need, S4 wouldn't match if data is missing or not DataFrame

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -2045,3 +2045,34 @@ setMethod("attach",
...
+            cols <- columns(data)
+            stopifnot(length(cols) > 0)
+
+            newEnv <- new.env()
+            for (i in 1:length(cols)) {
+              assign(x = cols[i], value = data[, cols[i]], envir = newEnv)
+            }
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: these few lines are exactly the same as in `attach`, is there an opportunity to share?

Comment (adrian555) [inline R/pkg/R/DataFrame.R]: Certainly I can create an internal function for this. Any rule and suggestion on where to put and what I should name the function? Thanks.

Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: typically we would add an internal S3 method in this file for helper function

Comment (shivaram) [inline R/pkg/R/DataFrame.R]: Or you can put it in utils.R

**Comment by felixcheung** [comment]
@shivaram

**Comment by shivaram** [comment]
LGTM. Thanks @adrian555

**Pull request closed without merging**